identity-iota/src/credential/validator.rs

```diff
@@ -138,11 +152,16 @@ impl<'a, R: TangleResolve> CredentialValidator<'a, R> {
 
     // Resolve and validate all associated credentials.
     for credential in presentation.verifiable_credential.iter() {
-      credentials.push(self.validate_credential(credential.clone()).await?);
+      // TODO: do we need to allow different VerifierOptions for each credential?
```

Comment:
If we get requests for such a feature in the future then we can perhaps add an additional method that also takes a closure `FnMut(&Credential)->Option<VerifierOptions>` to determine which  `VerifierOptions` to pass to `validate_credential`. For now I think it is fine the way it has been implemented.

Reply:
Good suggestion. Theoretically possible in the Wasm bindings too (https://rustwasm.github.io/wasm-bindgen/reference/receiving-js-closures-in-rust.html) but I think the problem there would be having to convert `Credential` into `WasmCredential` internally to be typed properly, which means duplicating the presentation validation code in the Wasm bindings? Not a major problem to be fair but I am still biased against closures because of the uncertainty with future bindings too.

Users can currently extract credentials from a presentation and re-validate them with the desired `VerifierOptions` if they want to, so we probably won't tackle this for a while unless we nail a really convenient API design.